### PR TITLE
Adding support for hash parameter to make

### PR DESCRIPTION
**Pull request by andrewguy9**
This patch allows me to write a plugin where I pass fields other than note and title, such as due_date.
By adding a new case, backwards compatibility with existing plugins is maintained.

### CODE REVIEW

File: lib/omnifocus.rb [hunk 1]
```diff
@@ -33,9 +33,9 @@ class OmniFocus
   ##
   # bug_db = {
   #   project => {
-  #     bts_id => [task_name, url], # only on BTS     = add to OF
-  #     bts_id => true,             # both BTS and OF = don't touch
-  #     bts_id => false,            # only on OF      = remove from OF
+  # bts_id => [task_name, url, due, defer], # only on BTS     = add to OF
+  # bts_id => {field=>value, ...},          # only on BTS     = OF and maybe BTS. Update fields
+  # bts_id => true,                         # both BTS and OF = don't touch
```
Comment (zenspider) [inline lib/omnifocus.rb]: Indentation is wrong. This is still sub-sub-hash.

File: lib/omnifocus.rb [hunk 2]
```diff
@@ -94,7 +94,8 @@ def all_tasks
   # Utility shortcut to make a new thing with a name via appscript.
 
   def make target, type, name, extra = {}
-    target.make :new => type, :with_properties => { :name => name }.merge(extra)
+    properties = { :name => name }.merge(extra)
+    target.make :new => type, :with_properties => properties
```
Comment (zenspider) [inline lib/omnifocus.rb]: This diff is unnecessary.

File: lib/omnifocus.rb [hunk 3]
```diff
@@ -170,15 +171,15 @@ def create_missing_projects
   # for +bug_db+ for more info on how you should populate it.
 
   def update_tasks
-    bug_db.each do |name, tickets|
-      project = nerd_projects.projects[name]
+    bug_db.each do |project_name, tickets|
+      project = nerd_projects.projects[project_name]
```
Comment (zenspider) [inline lib/omnifocus.rb]: These diffs are unnecessary to _this_ patch. But are welcome clarification. Can you split that commit out to its own PR?

File: lib/omnifocus.rb [hunk 4]
```diff
@@ -186,15 +187,21 @@ def update_tasks
...
             task.completed.set true
           end
         when Array
-          puts "Adding #{name} # #{bts_id}"
+          puts "Adding #{project_name} # #{bts_id}"
           next if $DEBUG
           title, url = *value
           make project, :task, title, :note => url
+        when Hash
+          puts "Adding Detail #{project_name} # #{bts_id} #{$DEBUG}"
```
Comment (zenspider) [inline lib/omnifocus.rb]: why `$DEBUG`?

**Comment by zenspider** [comment]
Looks pretty good! Thanks! Please address comments and I'll merge.

**Comment by andrewguy9** [comment]
Applied Feedback, care to take another look?

**Comment by andrewguy9** [comment]
Hey, did you have more feedback on this pull?

**Comment by zenspider** [comment]
> On Sep 12, 2015, at 16:51, Andrew Thomson <notifications@github.com> wrote:
> 
> Hey, did you have more feedback on this pull?

My first bit of feedback is to relax.

**Comment by zenspider** [comment]
Added. Thanks!
